### PR TITLE
Add md.obsidian.Obsidian (Obsidian)

### PR DESCRIPTION
**Pull request by joshua-stone**
This PR should fulfill a feature request made on the Obsidian community forum:

https://forum.obsidian.md/t/linux-packaging-ship-as-flatpak-on-flathub/6112

Note: I'm not one of the upstream developers.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build md.obsidian.Obsidian

**Comment by joshua-stone** [comment]
I've changed the flatpak to use the tarball instead of the appimage, which hopefully will let the build succeed this time.

**Comment by barthalion** [comment]
bot, build md.obsidian.Obsidian

**Comment by barthalion** [comment]
Build failed due to missing appdata file:

```
stat: cannot stat 'builddir/*/share/app-info/xmls/md.obsidian.Obsidian.xml.gz': No such file or directory
```

File: md.obsidian.Obsidian.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline md.obsidian.Obsidian.appdata.xml:1]: This file needs to be installed in the manifest.

File: md.obsidian.Obsidian.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+<component type="desktop">
+  <id>md.obsidian.Obsidian</id>
+  <launchable type="desktop-id">md.obsidian.Obsidian.desktop</launchable>
+  <name>Obsidian</name>
+  <summary>Markdown-based knowledge base</summary>
+  <description>
+    <p>Obsidian is a powerful knowledge base that works on top of a local folder of plain text Markdown files.</p>
```
Comment (barthalion) [inline md.obsidian.Obsidian.appdata.xml:8]: Could the description be expanded upon to include what the homepage says?

File: md.obsidian.Obsidian.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  <releases>
+    <release version="0.9.2" date="2020-09-30"/>
+  </releases>
+  <url type="homepage">https://obsidian.md/</url>
+  <url type="help">https://obsidian.md/community</url>
+  <url type="faq">https://obsidian.md/faq</url>
+  <url type="bugtracker">https://github.com/flathub/md.obsidian.Obsidian/issues</url>
+  <developer_name>Obsidian</developer_name>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary</project_license>
```
Comment (barthalion) [inline md.obsidian.Obsidian.appdata.xml:26]: If it's proprietary, the manifest should use `extra-data` to download it during installation, unless we are granted redistribution rights.

File: md.obsidian.Obsidian.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  - --socket=x11
+  - --device=dri
+  - --filesystem=home
+  - --share=network
+modules:
+  - name: obsidian
+    buildsystem: simple
+    build-commands:
+      - mkdir -p bin share/icons/hicolor/{16x16,32x32,48x48,64x64,128x128,256x256,512x512}/apps share/applications
+      - mv md.obsidian.Obsidian-16.png  share/icons/hicolor/16x16/apps/md.obsidian.Obsidian.png
```
Comment (barthalion) [inline md.obsidian.Obsidian.yml:21]: An 512x512 should be enough, it will be downscaled by appstream-util to smaller size for "internal" use.

**Comment by barthalion** [comment]
FYI you can trigger new builds yourself with `bot, build md.obsidian.Obsidian`.

**Comment by joshua-stone** [comment]
bot, build md.obsidian.Obsidian

**Comment by joshua-stone** [comment]
bot, build md.obsidian.Obsidian

File: md.obsidian.Obsidian.appdata.xml (hunk 3 above)
Comment (lishid) [reply md.obsidian.Obsidian.appdata.xml:26]: Obsidian maker here. Redistribution is fine as long as we can make it clear that the same EULA (https://obsidian.md/eula) applies to the end user. Anything we need to do on our end for this?

**Comment by joshua-stone** [comment]
bot, build md.obsidian.Obsidian

**Comment by joshua-stone** [comment]
bot, build md.obsidian.Obsidian

File: md.obsidian.Obsidian.appdata.xml (hunk 3 above)
Comment (barthalion) [reply md.obsidian.Obsidian.appdata.xml]: Thanks. Done via https://github.com/flathub/flathub/pull/1883/commits/0e61bc67faadd8a79e26aac452f988e352462608.

**Comment by barthalion** [comment]
My only gripe is short description. When that's improved, it's ready to merge.

**Comment by barthalion** [comment]
@lishid, would you like to have write access to the Flathub repository, just in case?

**Comment by lishid** [comment]
> @lishid, would you like to have write access to the Flathub repository, just in case?

Sure that would definitely help!

**Comment by joshua-stone** [comment]
bot, build md.obsidian.Obsidian

**Comment by joshua-stone** [comment]
bot, build md.obsidian.Obsidian

**Comment by joshua-stone** [comment]
bot, build md.obsidian.Obsidian

File: md.obsidian.Obsidian.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    build-commands:
+      - mkdir -p bin share/{appdata,applications,icons/hicolor/512x512/apps}
+      - mv md.obsidian.Obsidian.png share/icons/hicolor/512x512/apps/md.obsidian.Obsidian.png
+      - mv md.obsidian.Obsidian.desktop share/applications/md.obsidian.Obsidian.desktop
+      - mv obsidian.sh bin/obsidian.sh
+      - mv md.obsidian.Obsidian.appdata.xml share/appdata/md.obsidian.Obsidian.appdata.xml
+      - cp -r * ${FLATPAK_DEST}
+    sources:
+      - type: archive
+        url: https://github.com/obsidianmd/obsidian-releases/releases/download/v0.9.3/obsidian-0.9.3.tar.gz
```
Comment (barthalion) [inline md.obsidian.Obsidian.yml:28]: One thing I'd add here is `x-data-checker` as documented at https://github.com/flathub/flatpak-external-data-checker so Flathub version doesn't fall behind with releases, but that can be fixed after merge.

**Pull request closed without merging**